File: CSharp/DependencyBreakingTests/A/DiscountTest.cs
using Org.Codecop.Dependencies.A;
using Xunit;

namespace Org.Codecop.Dependencies.A.Tests
{
    public class DiscountTest
    {
        [Fact]
        public void Test1()
        {
            var discount = new Discount();

            var net = new Money(1002);
            var total = discount.DiscountFor(net);

            Assert.Equal(new Money(901.8m), total);
        }
    }
}

File: PHP/test/A/DiscountTest.php
<?php

namespace Org\Codecop\Dependencies\Tests\A;

use PHPUnit\Framework\TestCase;
use Brick\Math\BigDecimal;
use Org\Codecop\Dependencies\A\Discount;
use Org\Codecop\Dependencies\A\Money;

class DiscountTest extends TestCase {

    public function test1() {
        $discount = new Discount();

        $net = new Money(1002);
        $total = $discount->discountFor($net);

        $this->assertEquals(new Money(new BigDecimal("901.8")), $total);
    }
}
